pages/bookmarks.jsx

```diff
@@ -32,7 +33,12 @@ export default React.createClass({
   componentDidMount() {
     // get IDs of user's bookmarked entries
     this.setState({bookmarkedIds: getBookmarks()}, () => {
-      this.fetchData();
+      if (pageSettings.restore) {
+        // restore state back to what is stored in pageSettings
+        this.setState(Object.assign({},this.state,pageSettings.currentList));
```

Comment:
I don't quite understand this line. `setState` already copies over values from the "newstate" argument into whatever "this.state" is, so the `Object.assign` call here does the same as what `setState` already does. Was there a problem you ran into that necessitated this code?